server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamFailureStoreDefinition.java

```diff
@@ -27,6 +28,21 @@ public class DataStreamFailureStoreDefinition {
 
     public static final String FAILURE_STORE_REFRESH_INTERVAL_SETTING_NAME = "data_streams.failure_store.refresh_interval";
     public static final Settings DATA_STREAM_FAILURE_STORE_SETTINGS;
+    // Settings for removal, we do not use the `DATA_STREAM_FAILURE_STORE_SETTINGS` because
+    // some internal index settings cannot be set to null effectively.
+    public static final Set<String> UNSUPPORTED_SETTINGS = Set.of(
```

Comment:
We introduced here a denylist, but it feels a bit error prone, what if there are more settings interfering that we have listed here. 

@jbaiera & @dakrone do you think we could create (a reasonable in size) allowlist instead?

Reply:
Discussed offline, and we do prefer an allowlist. I will work on that.